src/fenics_constitutive/models/linear_elasticity_model.py

```diff
@@ -24,76 +26,26 @@ def __init__(
         E = parameters["E"]
         nu = parameters["nu"]
-        mu = E / (2.0 * (1.0 + nu))
-        lam = E * nu / ((1.0 + nu) * (1.0 - 2.0 * nu))
-        match constraint:
-            case StressStrainConstraint.FULL:
-                # see https://en.wikipedia.org/wiki/Hooke%27s_law
-                self.D = np.array(
-                    [
-                        [2.0 * mu + lam, lam, lam, 0.0, 0.0, 0.0],
-                        [lam, 2.0 * mu + lam, lam, 0.0, 0.0, 0.0],
-                        [lam, lam, 2.0 * mu + lam, 0.0, 0.0, 0.0],
-                        [0.0, 0.0, 0.0, 2.0 * mu, 0.0, 0.0],
-                        [0.0, 0.0, 0.0, 0.0, 2.0 * mu, 0.0],
-                        [0.0, 0.0, 0.0, 0.0, 0.0, 2.0 * mu],
-                    ]
-                )
-            case StressStrainConstraint.PLANE_STRAIN:
-                # We assert that the strain is being provided with 0 in the z-direction
-                # see https://en.wikipedia.org/wiki/Hooke%27s_law
-                self.D = np.array(
-                    [
-                        [2.0 * mu + lam, lam, lam, 0.0],
-                        [lam, 2.0 * mu + lam, lam, 0.0],
-                        [lam, lam, 2.0 * mu + lam, 0.0],
-                        [0.0, 0.0, 0.0, 2.0 * mu],
-                    ]
-                )
-            case StressStrainConstraint.PLANE_STRESS:
-                # We do not make any assumptions about strain in the z-direction
-                # This matrix just multiplies the z component by 0.0 which results
-                # in a plane stress state
-                # see https://en.wikipedia.org/wiki/Hooke%27s_law
-                self.D = (
-                    E
-                    / (1 - nu**2.0)
-                    * np.array(
-                        [
-                            [1.0, nu, 0.0, 0.0],
-                            [nu, 1.0, 0.0, 0.0],
-                            [0.0, 0.0, 0.0, 0.0],
-                            [0.0, 0.0, 0.0, (1.0 - nu)],
-                        ]
-                    )
-                )
-            case StressStrainConstraint.UNIAXIAL_STRAIN:
-                # see https://csmbrannon.net/2012/08/02/distinction-between-uniaxial-stress-and-uniaxial-strain/
-                C = E * (1.0 - nu) / ((1.0 + nu) * (1.0 - 2.0 * nu))
-                self.D = np.array([[C]])
-            case StressStrainConstraint.UNIAXIAL_STRESS:
-                # see https://csmbrannon.net/2012/08/02/distinction-between-uniaxial-stress-and-uniaxial-strain/
-                self.D = np.array([[E]])
-            case _:
-                msg = "Constraint not implemented"
-                raise NotImplementedError(msg)
+        law = get_elasticity_law(constraint)
+        self.D = law.get_D(E, nu)
 
     def evaluate(
         self,
-        time: float,
+        t: float,
         del_t: float,
         grad_del_u: np.ndarray,
-        mandel_stress: np.ndarray,
+        stress: np.ndarray,
```

Comment:
why would we remove the additional notation how the stress is defined? Same with t. IMO it improves readibility.